lib/index.js

```diff
@@ -13,5 +13,46 @@ module.exports = (dirname, manifestExtras) => {
 
     const manifest = Manifest.concat(manifestExtras || []);
 
-    return Haute(dirname, manifest);
+    return (server, options, next) => {
+
+        const promise = function () {
+
+            return new Promise((resolve, reject) => {
+
+                Haute(dirname, manifest)(server, options, (err) => {
+
+                    if (err) {
+                        reject(err);
+                    }
+                    else {
+                        resolve();
+                    }
+                });
+            });
+        };
+
+        if (typeof options !== 'undefined') {
+
+            // user sent next in second argument
+            if (typeof options === 'function') {
+
+                return Haute(dirname, manifest)(server, options, next);
```

Comment:
This is pretty odd– it's using `options` as a callback and `next` as an `undefined` but present argument.